src/main/java/com/google/maps/NearbySearchRequest.java

```diff
@@ -0,0 +1,160 @@
+import com.google.maps.model.*;
+
+public class NearbySearchRequest extends PendingResultBase<PlacesSearchResponse, NearbySearchRequest, NearbySearchRequest.Response> {
+
+  static final ApiConfig API_CONFIG = new ApiConfig("/maps/api/place/nearbysearch/json")
+      .fieldNamingPolicy(FieldNamingPolicy.LOWER_CASE_WITH_UNDERSCORES);
+
+  public NearbySearchRequest(GeoApiContext context) {
+    super(context, API_CONFIG, Response.class);
+  }
+
+  /**
+   * location is the latitude/longitude around which to retrieve place information.
+   */
+  public NearbySearchRequest location(LatLng location) {
+    return param("location", location);
+  }
+
+  /**
+   * radius defines the DISTANCE (in meters) within which to return place results. The maximum allowed radius is
+   * 50,000 meters. Note that radius must not be included if rankby=DISTANCE is specified.
+   */
+  public NearbySearchRequest radius(int distance) {
+    return param("radius", Integer.toString(distance));
+  }
+
+  /**
+   * rankby specifies the order in which results are listed.
+   */
+  public NearbySearchRequest rankby(Rankby ranking) {
+    return param("rankby", ranking);
+  }
+
+  /**
+   * keyword is a term to be matched against all content that Google has indexed for this place, including but not
+   * limited to name, type, and address, as well as customer reviews and other third-party content.
+   */
+  public NearbySearchRequest keyword(String keyword) {
+    return param("keyword", keyword);
+  }
+
+  /**
+   * minPrice restricts to places that are at least this price level.
+   */
+  public NearbySearchRequest minPrice(PriceLevel priceLevel) {
+    return param("minprice", priceLevel);
+  }
+
+  /**
+   * maxPrice restricts to places that are at most this price level.
+   */
+  public NearbySearchRequest maxPrice(PriceLevel priceLevel) {
+    return param("maxprice", priceLevel);
+  }
+
+  /**
+   * name is one or more terms to be matched against the names of places, separated with a space character.
+   */
+  public NearbySearchRequest name(String name) {
+    return param("name", name);
+  }
+
+  /**
+   * openNow returns only those places that are open for business at the time the query is sent.
+   */
+  public NearbySearchRequest openNow(boolean openNow) {
+    return param("opennow", String.valueOf(openNow));
+  }
+
+  /**
+   * pageToken returns the next 20 results from a previously run search. Setting a pageToken parameter will execute a
+   * search with the same parameters used previously — all parameters other than pageToken will be ignored.
+   */
+  public NearbySearchRequest pageToken(String nextPageToken) {
+    return param("pagetoken", nextPageToken);
+  }
+
+  /**
+   * type ...
+   */
+  public NearbySearchRequest type(PlaceType type) {
+    return param("type", type);
+  }
+
+  @Override
+  protected void validateRequest() {
+
+    // If pagetoken is included, all other parameters are ignored.
+    if (!params().containsKey("pagetoken")) {
```

Comment:
It's up to you, but I think this would be more readable-

```java
params = params();
if (params.containsKey("pagetoken")) {
  return;  // if pagetoken is included, all other parameters are ignored
}
// rest
```